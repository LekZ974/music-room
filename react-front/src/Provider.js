import React from 'react';

import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { AppLoading } from 'expo';
import store, { persistor } from './redux/store';

import Main from './Main';
import { alert } from './redux/actions/alert';

const WrapperProvider = () => (
  <Provider store={store}>
    <PersistGate loading={<AppLoading />} persistor={persistor}>
      <Main />
    </PersistGate>
  </Provider>
);

store.dispatch(alert('Soon, will be here a fantastic Music Room ...'));
export default WrapperProvider;
