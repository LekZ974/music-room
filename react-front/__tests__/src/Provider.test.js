import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Provider from '../../src/Provider';

jest.mock('react-native-gesture-handler', () => {});
jest.mock('../../src/i18n', () => {});
jest.mock('../../src/routes.js', () => {});

describe('Provider snapshot', () => {
  it('renders the loading screen', async () => {
    let wrapper = shallow(<Provider />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});