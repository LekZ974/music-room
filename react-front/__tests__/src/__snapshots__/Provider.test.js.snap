// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider snapshot renders the loading screen 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={<AppLoading />}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(lifecycle(Component)) />
  </PersistGate>
</Provider>
`;
